httpMiddleware.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/palantir/go-githubapp/githubapp"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/rs/zerolog"
+)
+
+type statusCodeResponseWriter struct {
+	http.ResponseWriter
+	statusCode int
+}
+
+func (w *statusCodeResponseWriter) WriteHeader(code int) {
+	w.statusCode = code
+	w.ResponseWriter.WriteHeader(code)
+}
+func InitializeResponder(ctx context.Context) context.Context {
+	var responder func(http.ResponseWriter, *http.Request)
+	return context.WithValue(ctx, responderKey{}, &responder)
+}
+
+type responderKey struct{}
+type LoggerFunc func(msg string, m map[string]interface{})
+
+func addContextMiddleware(logger zerolog.Logger, h http.Handler) http.Handler {
```

Comment:
Maybe the name of this can be more explicit? It adds a logger to the request context, right? So maybe `loggingContextMiddleware` or something like that?

Reply:
https://github.com/lunarway/release-manager-bot/pull/21/commits/effead85e30c743cbfd1f19fce43ec73086b5fc9